pennylane/variable.py

```diff
@@ -94,12 +92,12 @@ class VariableRef:
     #: dict[str->array[float]]: current auxiliary parameter values, set in :meth:`.BaseQNode._set_variables`
     kwarg_values = None
 
-    def __init__(self, idx, name=None, is_kwarg=False):
+    def __init__(self, idx, name=None, basename=None):
```

Comment:
What is the difference between name and basename? Basename should be added to the Args: list.

Reply:
So, for VariableRefs that represent kwargs, basename is the name of the kwarg, and name is the "structured variable name". For example if we have a kwarg named 'x' that is an array, each of its elements is represented by a separate VariableRef. We would have one with basename='x', name='x[0][0]' etc.

Reply:
Oh nice! would be worth adding this as a comment for clarity.

Reply:
But shouldn't args also have a basename? You could very well have `weights` as arg and `weights[0][0]` as name. In this case I find the name `basename` confusing and would rather call it `dict_key` or something like this.

Reply:
Args could also have a basename like you said, but we don't have a use for it at the moment. dict_key could be less confusing.

Reply:
Now it's dict_key.